### PR TITLE
layerscape: support sysupgrade

### PR DESCRIPTION
**Pull request by yangbolu1991**
This patch-set is to support sysupgrade for both squashfs rootfs on flash, and ext4 rootfs on SD card.
Please help to review and merge.
Thanks a lot!

### CODE REVIEW

**Comment by yangbolu1991** [comment]
I added more patches to support sysupgrade, and updated pull request title and description.
Thanks a lot:)

**Comment by yangbolu1991** [comment]
Any comments?
Thanks!

**Comment by yangbolu1991** [comment]
Hi @ynezz  
Will you help to review the pull request? There has been no feedback for a long time since created.
Thanks a lot!

**Comment by adrianschmutzler** [comment]
> Backport a dts patch which added compatible for ls1043ardb.

If that's a backport from the kernel, it might help to include the kernel version where it was added in the _patch_ file name, e.g. `724-v5.5-net-sfp-rename-T_PROBE_WAIT-to-T_SERIAL.patch`

File: target/linux/layerscape/image/armv8_64b.mk [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 define Device/Default
   PROFILES := Default
-  IMAGES := firmware.bin
+  IMAGES := firmware.bin sysupgrade.bin
   FILESYSTEMS := squashfs
   KERNEL := kernel-bin | gzip | uImage gzip
   KERNEL_LOADADDR := 0x80080000
   KERNEL_ENTRY_POINT := 0x80080000
+  IMAGE/sysupgrade.bin := append-kernel | pad-to 16M | \
+    append-rootfs | pad-rootfs | check-size 50331649 | append-metadata
```
Comment (adrianschmutzler) [inline target/linux/layerscape/image/armv8_64b.mk:16]: Where is the argument for check-size coming from?
Consider defining IMAGE_SIZE instead.

**Comment by ynezz** [comment]
@tmn505 @mcbridematt can you please check/test this?

**Comment by yangbolu1991** [comment]
Here are two logs for flash sysupgrade, and SD sysupgarde on ls1088ardb, I verified that time.
Just for your reference. I should have attached them.
Thanks.

[openwrt-202005-ls1088ardb-sysupgrade-ext4.txt](https://github.com/openwrt/openwrt/files/4888481/openwrt-202005-ls1088ardb-sysupgrade-ext4.txt)
[openwrt-202005-ls1088ardb-sysupgrade-squashfs.txt](https://github.com/openwrt/openwrt/files/4888482/openwrt-202005-ls1088ardb-sysupgrade-squashfs.txt)

**Comment by tmn505** [comment]
It looks ok. The only downside with this is, that once written dtb will remain the same after every sysupgrade. To remedy this, I would suggest using FIT image.

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

File: target/linux/layerscape/image/armv8_64b.mk [hunk 2]
```diff
@@ -86,6 +86,12 @@ define Device/ls1012afrwy
     ls-append-dtb $$(DEVICE_DTS) | pad-to 16M | \
     append-kernel | pad-to $(LS_SD_ROOTFSPART_OFFSET)M | \
     append-rootfs | check-size $(LS_SD_IMAGE_SIZE)
+  IMAGE/sysupgrade.bin := \
+    ls-clean | \
+    ls-append-sdhead $(1) | pad-to 16M | \
+    append-kernel | pad-to $(LS_SD_ROOTFSPART_OFFSET)M | \
+    append-rootfs | check-size $(LS_SD_IMAGE_SIZE) | append-metadata
+  SUPPORTED_DEVICES := fsl,ls1012a-frwy-sdboot
```
Comment (adrianschmutzler) [inline target/linux/layerscape/image/armv8_64b.mk:97]: I don't find these -sdboot suffixed values anywhere else, particularly not in any compatible as you partially reuse the DTS files anywhere. Is this actually working for sysupgrade then?

Comment (yangbolu1991) [reply target/linux/layerscape/image/armv8_64b.mk:97]: In OpenWrt, the board name info is created in by /lib/preinit/02_sysinfo.
You can find below patch of this pull request to fix up the board name for SD card boot.
layerscape: fix board name for SD card boot

Comment (adrianschmutzler) [reply target/linux/layerscape/image/armv8_64b.mk:97]: Okay, this was the missing link I was looking. While I don't really like that approach conceptually, it appears to do the job.

Can you explain to me how the filtering works there? How do you make sure -sdboot is only added for sdboot variants, and not for all devices.

(Despite, I'd have used a higher number for that script, as sorting between 02_sysinfo and 02_sysinfo_fixup is not really obvious).

**Pull request closed without merging**